plugins/woocommerce/includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -392,4 +487,115 @@ protected function prepare_column_headers() {
 			$this->get_sortable_columns(),
 		);
 	}
+
+	/**
+	 * Convert an array or object into a multidimensional array.
+	 *
+	 * @param array|object $input Array or object to be processed.
+	 *
+	 * @return array
+	 */
+	public function to_multidimensional_array( $input ) {
+		$out   = array();
+		$input = is_object( $input ) ? (array) $input : $input;
+
+		if ( is_array( $input ) ) {
+
+			// Run function on each item in the array.
+			$out = array_map(
+				function( $value ) {
+
+					if ( is_array( $value ) || is_object( $value ) ) {
+						return $this->to_multidimensional_array( $value );
+					}
+
+					return $value;
+				},
+				$input
+			);
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Convert an array to a string with tabbed indentaion.
+	 *
+	 * @param array|object $input   An object or array.
+	 * @param array        $exclude An array of strings of params/keys to exclude.
+	 * @param int          $depth   The current depth of the walker.
+	 *
+	 * @return string
+	 */
+	public function printable_array( $input, $exclude = array(), $depth = 0 ) {
+		$out   = '';
+		$input = is_object( $input ) ? (array) $input : $input;
+		if ( is_array( $input ) ) {
+
+			foreach ( $input as $key => $val ) {
+
+				$key = strip_tags( (string) $key );
+
+				if ( in_array( (string) $key, $exclude ) ) {
+					continue;
+				}
+
+				$tabs = '';
+				for ( $i = 0; $i < $depth; $i++ ) {
+					$tabs .= '  ';
+				}
+
+				$out .= $tabs . '[' . $key . '] => ';
+
+				if ( is_array( $val ) || is_object( $val ) ) {
+					$new_depth = $depth + 1;
+					$out .= '[&#010;' . $this->printable_array( $val, $exclude, $new_depth ) . $tabs . ']';
+				} else {
+					$out .= "'" . $val . "'";
+				}
```

Comment:
Possible alternative:

```suggestion
				$out .= is_array( $val )
					? "[\n" . $this->printable_array( $val, $exclude, $depth + 1 ) . $tabs . ']'
					: "'$val'";
```

Thinking here that the `is_object()` check is not needed, as we convert objects to arrays at an earlier stage, and the creation of var `$new_depth` probably doesn't add much readability over `$depth + 1`. We could also apply these thoughts to the existing `if/else` structure, it doesn't need to be changed into a ternary expression.

Reply:
I tested this to be sure, and it's possible the array being looped through has objects as values.  Without the `is_object()` check here, things like the `WC_Meta_Data` object cause major problems.

I suppose the `multidimensional_array()` method I removed would have prevented this, as it would walk the entire depth of the array.  However, I think it's much more efficient and versatile to just keep this check here inside `printable_array()`.